app/controllers/events_controller.rb

```diff
@@ -48,19 +48,29 @@ def edit
   def create
     @event = Event.new(event_params)
 
+    if params[:save_draft]
+      @event.draft_saved = true
+    end
+
     if @event.save
       @event.organizers << current_user
-      if current_user.spammer?
-        @event.update_attribute(:spam, true)
-      else
-        EventMailer.unpublished_event(@event).deliver_now
-        EventMailer.event_pending_approval(@event).deliver_now
-      end
 
-      if @event.published
-        redirect_to @event, notice: 'Event was successfully created.'
-      else
+      case @event.current_state
+      when nil # No state set => it's now pending approval
```

Comment:
I wonder if having an explicit state for pending approval would be more clear than relying on a lack of one?